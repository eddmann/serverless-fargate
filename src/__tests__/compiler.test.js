const compile = require('../compiler');

test('single service task', () => {
  const compiled = compile(
    {
      'my-image': 'image-uri',
    },
    {
      memory: '0.5GB',
      cpu: 256,
      environment: {
        provider: 'variable',
      },
      iamRoleStatements: [
        {
          Effect: 'Allow',
          Action: ['sqs:*'],
          Resource: '*',
        },
      ],
      iamManagedPolicies: ['arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess'],
      vpc: {
        subnetIds: ['subnet-1234', 'subnet-5678'],
        securityGroupIds: ['sg-1234'],
        assignPublicIp: false,
      },
      tags: {
        provider: 'tag',
      },
      tasks: [
        {
          name: 'my-task',
          image: 'my-image',
          vpc: {
            subnetIds: ['subnet-1234', 'subnet-5678'],
            securityGroupIds: ['sg-1234'],
            assignPublicIp: false,
          },
          command: ['command'],
          entryPoint: ['entrypoint'],
          memory: '0.5GB',
          cpu: 256,
          environment: {
            task: 'variable',
          },
          tags: {
            task: 'tag',
          },
          service: {
            desiredCount: 1,
            maximumPercent: 200,
            minimumHealthyPercent: 100,
            spot: false,
          },
        },
      ],
    }
  );

  expect(compiled).toMatchSnapshot();
});

test('single scheduled task', () => {
  const compiled = compile(
    {
      'my-image': 'image-uri',
    },
    {
      memory: '0.5GB',
      cpu: 256,
      environment: {
        provider: 'variable',
      },
      iamRoleStatements: [
        {
          Effect: 'Allow',
          Action: ['sqs:*'],
          Resource: '*',
        },
      ],
      iamManagedPolicies: ['arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess'],
      vpc: {
        subnetIds: ['subnet-1234', 'subnet-5678'],
        securityGroupIds: ['sg-1234'],
        assignPublicIp: false,
      },
      tags: {
        provider: 'tag',
      },
      tasks: [
        {
          name: 'my-task',
          image: 'my-image',
          vpc: {
            subnetIds: ['subnet-1234', 'subnet-5678'],
            securityGroupIds: ['sg-1234'],
            assignPublicIp: false,
          },
          command: ['command'],
          entryPoint: ['entrypoint'],
          memory: '0.5GB',
          cpu: 256,
          environment: {
            task: 'variable',
          },
          tags: {
            task: 'tag',
          },
          schedule: 'rate(1 minute)',
        },
      ],
    }
  );

  expect(compiled).toMatchSnapshot();
});

test('service and scheduled tasks', () => {
  const compiled = compile(
    {
      'my-image-1': 'image-uri-1',
      'my-image-2': 'image-uri-2',
    },
    {
      memory: '0.5GB',
      cpu: 256,
      environment: {
        provider: 'variable',
      },
      iamRoleStatements: [
        {
          Effect: 'Allow',
          Action: ['sqs:*'],
          Resource: '*',
        },
      ],
      iamManagedPolicies: ['arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess'],
      vpc: {
        subnetIds: ['subnet-1234', 'subnet-5678'],
        securityGroupIds: ['sg-1234'],
        assignPublicIp: false,
      },
      tags: {
        provider: 'tag',
      },
      tasks: [
        {
          name: 'my-task-1',
          image: 'my-image-1',
          vpc: {
            subnetIds: ['subnet-1234', 'subnet-5678'],
            securityGroupIds: ['sg-1234'],
            assignPublicIp: false,
          },
          command: ['command'],
          entryPoint: ['entrypoint'],
          memory: '0.5GB',
          cpu: 256,
          environment: {
            task: 'variable',
          },
          tags: {
            task: 'tag',
          },
          service: {
            desiredCount: 1,
            maximumPercent: 200,
            minimumHealthyPercent: 100,
            spot: false,
          },
        },
        {
          name: 'my-task-2',
          image: 'my-image-2',
          vpc: {
            subnetIds: ['subnet-1234', 'subnet-5678'],
            securityGroupIds: ['sg-1234'],
            assignPublicIp: false,
          },
          command: ['command'],
          entryPoint: ['entrypoint'],
          memory: '0.5GB',
          cpu: 256,
          environment: {
            task: 'variable',
          },
          tags: {
            task: 'tag',
          },
          schedule: 'rate(1 minute)',
        },
      ],
    }
  );

  expect(compiled).toMatchSnapshot();
});

test('definition without IAM statements/policies present', () => {
  const compiled = compile(
    {},
    {
      memory: '0.5GB',
      cpu: 256,
      environment: {},
      iamRoleStatements: [],
      iamManagedPolicies: [],
      vpc: {
        subnetIds: ['subnet-1234', 'subnet-5678'],
        securityGroupIds: ['sg-1234'],
        assignPublicIp: false,
      },
      tags: {},
      tasks: [],
    }
  );

  const role = compiled.Resources.FargateIamTaskRole.Properties;
  expect(role.Policies).toEqual([]);
  expect(role.ManagedPolicyArns).toEqual([]);

  expect(compiled).toMatchSnapshot();
});

test('single service task using ec2 provider', () => {
  const compiled = compile(
    {
      'my-image': 'image-uri',
    },
    {
      autoScalingGroupCapacityProviderName: 'my-asg-capacity-provider',
      memory: '0.5GB',
      cpu: 256,
      environment: {
        provider: 'variable',
      },
      iamRoleStatements: [
        {
          Effect: 'Allow',
          Action: ['sqs:*'],
          Resource: '*',
        },
      ],
      iamManagedPolicies: ['arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess'],
      vpc: {
        subnetIds: ['subnet-1234', 'subnet-5678'],
        securityGroupIds: ['sg-1234'],
        assignPublicIp: false,
      },
      tags: {
        provider: 'tag',
      },
      tasks: [
        {
          autoScalingGroupCapacityProviderName: 'my-asg-capacity-provider',
          name: 'my-task',
          image: 'my-image',
          vpc: {
            subnetIds: ['subnet-1234', 'subnet-5678'],
            securityGroupIds: ['sg-1234'],
            assignPublicIp: false,
          },
          command: ['command'],
          entryPoint: ['entrypoint'],
          memory: '0.5GB',
          cpu: 256,
          environment: {
            task: 'variable',
          },
          tags: {
            task: 'tag',
          },
          service: {
            desiredCount: 1,
            maximumPercent: 200,
            minimumHealthyPercent: 100,
            spot: false,
          },
        },
      ],
    }
  );

  expect(compiled).toMatchSnapshot();
});

test('single scheduled task using ec2 provider', () => {
  const compiled = compile(
    {
      'my-image': 'image-uri',
    },
    {
      autoScalingGroupCapacityProviderName: 'my-asg-capacity-provider',
      memory: '0.5GB',
      cpu: 256,
      environment: {
        provider: 'variable',
      },
      iamRoleStatements: [
        {
          Effect: 'Allow',
          Action: ['sqs:*'],
          Resource: '*',
        },
      ],
      iamManagedPolicies: ['arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess'],
      vpc: {
        subnetIds: ['subnet-1234', 'subnet-5678'],
        securityGroupIds: ['sg-1234'],
        assignPublicIp: false,
      },
      tags: {
        provider: 'tag',
      },
      tasks: [
        {
          autoScalingGroupCapacityProviderName: 'my-asg-capacity-provider',
          name: 'my-task',
          image: 'my-image',
          vpc: {
            subnetIds: ['subnet-1234', 'subnet-5678'],
            securityGroupIds: ['sg-1234'],
            assignPublicIp: false,
          },
          command: ['command'],
          entryPoint: ['entrypoint'],
          memory: '0.5GB',
          cpu: 256,
          environment: {
            task: 'variable',
          },
          tags: {
            task: 'tag',
          },
          schedule: 'rate(1 minute)',
        },
      ],
    }
  );

  expect(compiled).toMatchSnapshot();
});
