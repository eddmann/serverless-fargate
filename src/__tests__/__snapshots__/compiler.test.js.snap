// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`definition without IAM statements/policies present 1`] = `
Object {
  "Outputs": Object {},
  "Resources": Object {
    "FargateIamExecutionRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                  "events.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy",
          "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceEventsRole",
        ],
        "Tags": Array [],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateIamTaskRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [],
        "Policies": Array [],
        "Tags": Array [],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateTasksCluster": Object {
      "Properties": Object {
        "CapacityProviders": Array [
          "FARGATE",
          "FARGATE_SPOT",
        ],
        "ClusterName": undefined,
        "Tags": Array [],
      },
      "Type": "AWS::ECS::Cluster",
    },
    "FargateTasksLogGroup": Object {
      "Properties": Object {
        "LogGroupName": undefined,
        "Tags": Array [],
      },
      "Type": "AWS::Logs::LogGroup",
    },
  },
}
`;

exports[`service and scheduled tasks 1`] = `
Object {
  "Outputs": Object {
    "Mytask1ServiceArn": Object {
      "Value": Object {
        "Ref": "Mytask1Service",
      },
    },
    "Mytask1TaskArn": Object {
      "Value": Object {
        "Ref": "Mytask1Task",
      },
    },
    "Mytask2ScheduledTaskArn": Object {
      "Value": Object {
        "Fn::GetAtt": Array [
          "Mytask2ScheduledTask",
          "Arn",
        ],
      },
    },
    "Mytask2TaskArn": Object {
      "Value": Object {
        "Ref": "Mytask2Task",
      },
    },
  },
  "Resources": Object {
    "FargateIamExecutionRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                  "events.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy",
          "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceEventsRole",
        ],
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateIamTaskRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess",
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "sqs:*",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "FargateTaskPolicy",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateTasksCluster": Object {
      "Properties": Object {
        "CapacityProviders": Array [
          "FARGATE",
          "FARGATE_SPOT",
        ],
        "ClusterName": undefined,
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::ECS::Cluster",
    },
    "FargateTasksLogGroup": Object {
      "Properties": Object {
        "LogGroupName": undefined,
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::Logs::LogGroup",
    },
    "Mytask1Service": Object {
      "Properties": Object {
        "CapacityProviderStrategy": Array [
          Object {
            "CapacityProvider": "FARGATE",
            "Weight": 1,
          },
        ],
        "Cluster": Object {
          "Fn::Sub": "\${FargateTasksCluster}",
        },
        "DeploymentConfiguration": Object {
          "MaximumPercent": 200,
          "MinimumHealthyPercent": 100,
        },
        "DesiredCount": 1,
        "NetworkConfiguration": Object {
          "AwsvpcConfiguration": Object {
            "AssignPublicIp": "DISABLED",
            "SecurityGroups": Array [
              "sg-1234",
            ],
            "Subnets": Array [
              "subnet-1234",
              "subnet-5678",
            ],
          },
        },
        "PropagateTags": "TASK_DEFINITION",
        "ServiceName": "my-task-1",
        "Tags": Array [
          Object {
            "Key": "task",
            "Value": "tag",
          },
        ],
        "TaskDefinition": Object {
          "Fn::Sub": "\${Mytask1Task}",
        },
      },
      "Type": "AWS::ECS::Service",
    },
    "Mytask1Task": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Command": Array [
              "command",
            ],
            "EntryPoint": Array [
              "entrypoint",
            ],
            "Environment": Array [
              Object {
                "Name": "task",
                "Value": "variable",
              },
            ],
            "Image": undefined,
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Fn::Sub": "\${FargateTasksLogGroup}",
                },
                "awslogs-region": Object {
                  "Fn::Sub": "\${AWS::Region}",
                },
                "awslogs-stream-prefix": "fargate",
              },
            },
            "Name": "my-task-1",
          },
        ],
        "Cpu": 256,
        "ExecutionRoleArn": Object {
          "Fn::Sub": "\${FargateIamExecutionRole}",
        },
        "Family": "my-task-1",
        "Memory": "0.5GB",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "Tags": Array [
          Object {
            "Key": "task",
            "Value": "tag",
          },
        ],
        "TaskRoleArn": Object {
          "Fn::Sub": "\${FargateIamTaskRole}",
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "Mytask2ScheduledTask": Object {
      "Properties": Object {
        "ScheduleExpression": "rate(1 minute)",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "FargateTasksCluster",
                "Arn",
              ],
            },
            "EcsParameters": Object {
              "LaunchType": "FARGATE",
              "NetworkConfiguration": Object {
                "AwsVpcConfiguration": Object {
                  "AssignPublicIp": "DISABLED",
                  "SecurityGroups": Array [
                    "sg-1234",
                  ],
                  "Subnets": Array [
                    "subnet-1234",
                    "subnet-5678",
                  ],
                },
              },
              "TaskCount": 1,
              "TaskDefinitionArn": Object {
                "Fn::Sub": "\${Mytask2Task}",
              },
            },
            "Id": "Mytask2",
            "RoleArn": Object {
              "Fn::GetAtt": Array [
                "FargateIamExecutionRole",
                "Arn",
              ],
            },
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "Mytask2Task": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Command": Array [
              "command",
            ],
            "EntryPoint": Array [
              "entrypoint",
            ],
            "Environment": Array [
              Object {
                "Name": "task",
                "Value": "variable",
              },
            ],
            "Image": undefined,
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Fn::Sub": "\${FargateTasksLogGroup}",
                },
                "awslogs-region": Object {
                  "Fn::Sub": "\${AWS::Region}",
                },
                "awslogs-stream-prefix": "fargate",
              },
            },
            "Name": "my-task-2",
          },
        ],
        "Cpu": 256,
        "ExecutionRoleArn": Object {
          "Fn::Sub": "\${FargateIamExecutionRole}",
        },
        "Family": "my-task-2",
        "Memory": "0.5GB",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "Tags": Array [
          Object {
            "Key": "task",
            "Value": "tag",
          },
        ],
        "TaskRoleArn": Object {
          "Fn::Sub": "\${FargateIamTaskRole}",
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
  },
}
`;

exports[`single scheduled task 1`] = `
Object {
  "Outputs": Object {
    "MytaskScheduledTaskArn": Object {
      "Value": Object {
        "Fn::GetAtt": Array [
          "MytaskScheduledTask",
          "Arn",
        ],
      },
    },
    "MytaskTaskArn": Object {
      "Value": Object {
        "Ref": "MytaskTask",
      },
    },
  },
  "Resources": Object {
    "FargateIamExecutionRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                  "events.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy",
          "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceEventsRole",
        ],
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateIamTaskRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess",
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "sqs:*",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "FargateTaskPolicy",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateTasksCluster": Object {
      "Properties": Object {
        "CapacityProviders": Array [
          "FARGATE",
          "FARGATE_SPOT",
        ],
        "ClusterName": undefined,
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::ECS::Cluster",
    },
    "FargateTasksLogGroup": Object {
      "Properties": Object {
        "LogGroupName": undefined,
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::Logs::LogGroup",
    },
    "MytaskScheduledTask": Object {
      "Properties": Object {
        "ScheduleExpression": "rate(1 minute)",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "FargateTasksCluster",
                "Arn",
              ],
            },
            "EcsParameters": Object {
              "LaunchType": "FARGATE",
              "NetworkConfiguration": Object {
                "AwsVpcConfiguration": Object {
                  "AssignPublicIp": "DISABLED",
                  "SecurityGroups": Array [
                    "sg-1234",
                  ],
                  "Subnets": Array [
                    "subnet-1234",
                    "subnet-5678",
                  ],
                },
              },
              "TaskCount": 1,
              "TaskDefinitionArn": Object {
                "Fn::Sub": "\${MytaskTask}",
              },
            },
            "Id": "Mytask",
            "RoleArn": Object {
              "Fn::GetAtt": Array [
                "FargateIamExecutionRole",
                "Arn",
              ],
            },
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "MytaskTask": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Command": Array [
              "command",
            ],
            "EntryPoint": Array [
              "entrypoint",
            ],
            "Environment": Array [
              Object {
                "Name": "task",
                "Value": "variable",
              },
            ],
            "Image": undefined,
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Fn::Sub": "\${FargateTasksLogGroup}",
                },
                "awslogs-region": Object {
                  "Fn::Sub": "\${AWS::Region}",
                },
                "awslogs-stream-prefix": "fargate",
              },
            },
            "Name": "my-task",
          },
        ],
        "Cpu": 256,
        "ExecutionRoleArn": Object {
          "Fn::Sub": "\${FargateIamExecutionRole}",
        },
        "Family": "my-task",
        "Memory": "0.5GB",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "Tags": Array [
          Object {
            "Key": "task",
            "Value": "tag",
          },
        ],
        "TaskRoleArn": Object {
          "Fn::Sub": "\${FargateIamTaskRole}",
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
  },
}
`;

exports[`single service task 1`] = `
Object {
  "Outputs": Object {
    "MytaskServiceArn": Object {
      "Value": Object {
        "Ref": "MytaskService",
      },
    },
    "MytaskTaskArn": Object {
      "Value": Object {
        "Ref": "MytaskTask",
      },
    },
  },
  "Resources": Object {
    "FargateIamExecutionRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                  "events.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy",
          "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceEventsRole",
        ],
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateIamTaskRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ecs-tasks.amazonaws.com",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess",
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "sqs:*",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "FargateTaskPolicy",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FargateTasksCluster": Object {
      "Properties": Object {
        "CapacityProviders": Array [
          "FARGATE",
          "FARGATE_SPOT",
        ],
        "ClusterName": undefined,
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::ECS::Cluster",
    },
    "FargateTasksLogGroup": Object {
      "Properties": Object {
        "LogGroupName": undefined,
        "Tags": Array [
          Object {
            "Key": "provider",
            "Value": "tag",
          },
        ],
      },
      "Type": "AWS::Logs::LogGroup",
    },
    "MytaskService": Object {
      "Properties": Object {
        "CapacityProviderStrategy": Array [
          Object {
            "CapacityProvider": "FARGATE",
            "Weight": 1,
          },
        ],
        "Cluster": Object {
          "Fn::Sub": "\${FargateTasksCluster}",
        },
        "DeploymentConfiguration": Object {
          "MaximumPercent": 200,
          "MinimumHealthyPercent": 100,
        },
        "DesiredCount": 1,
        "NetworkConfiguration": Object {
          "AwsvpcConfiguration": Object {
            "AssignPublicIp": "DISABLED",
            "SecurityGroups": Array [
              "sg-1234",
            ],
            "Subnets": Array [
              "subnet-1234",
              "subnet-5678",
            ],
          },
        },
        "PropagateTags": "TASK_DEFINITION",
        "ServiceName": "my-task",
        "Tags": Array [
          Object {
            "Key": "task",
            "Value": "tag",
          },
        ],
        "TaskDefinition": Object {
          "Fn::Sub": "\${MytaskTask}",
        },
      },
      "Type": "AWS::ECS::Service",
    },
    "MytaskTask": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Command": Array [
              "command",
            ],
            "EntryPoint": Array [
              "entrypoint",
            ],
            "Environment": Array [
              Object {
                "Name": "task",
                "Value": "variable",
              },
            ],
            "Image": undefined,
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Fn::Sub": "\${FargateTasksLogGroup}",
                },
                "awslogs-region": Object {
                  "Fn::Sub": "\${AWS::Region}",
                },
                "awslogs-stream-prefix": "fargate",
              },
            },
            "Name": "my-task",
          },
        ],
        "Cpu": 256,
        "ExecutionRoleArn": Object {
          "Fn::Sub": "\${FargateIamExecutionRole}",
        },
        "Family": "my-task",
        "Memory": "0.5GB",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "Tags": Array [
          Object {
            "Key": "task",
            "Value": "tag",
          },
        ],
        "TaskRoleArn": Object {
          "Fn::Sub": "\${FargateIamTaskRole}",
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
  },
}
`;
